app/utils/deep_link/index.ts

```diff
@@ -114,37 +120,66 @@ export async function handleDeepLink(deepLinkUrl: string, intlShape?: IntlShape,
     }
 }
 
+const matcherOpts: Parameters<typeof match>[1] = {decode: decodeURIComponent};
+
+type ChannelPathParams = {
+    hostname: string;
+    subpath?: string;
+    teamName: string;
+    path: 'channels' | 'messages';
+    identifier: string;
+    postId?: string;
+};
+
+const CHANNEL_PATH = `/:subpath?/:teamName(${TEAM_NAME_PATH_PATTERN})/:path(channels|messages)/:identifier(${IDENTIFIER_PATH_PATTERN}|${ID_PATH_PATTERN})/:postId(${ID_PATH_PATTERN})?`;
+export const matchChannelDeeplink = match<ChannelPathParams>(CHANNEL_PATH, matcherOpts);
+
+type PermalinkPathParams = {
+    hostname: string;
+    subpath?: string;
+    teamName: string;
+    postId: string;
+};
+const PERMALINK_PATH = `/:subpath?/:teamName(${TEAM_NAME_PATH_PATTERN})/pl/:postId(${ID_PATH_PATTERN})`;
+export const matchPermalinkDeeplink = match<PermalinkPathParams>(PERMALINK_PATH, matcherOpts);
+
+const getServerUrl = (host: string, subpath?: string) => host + (subpath ? '/' + subpath : '');
+
 export function parseDeepLink(deepLinkUrl: string): DeepLinkWithData {
     try {
-        const url = removeProtocol(decodeURIComponent(deepLinkUrl));
+        const {host, pathname} = urlParse(decodeURIComponent(deepLinkUrl));
```

Comment:
Is the `decodeURIComponent` still necessary?

Reply:
removed